internal/deviations/deviations.go

```diff
@@ -66,9 +66,37 @@ package deviations
 import (
 	"flag"
 
+	log "github.com/golang/glog"
+	"github.com/openconfig/featureprofiles/internal/metadata"
+	mpb "github.com/openconfig/featureprofiles/proto/metadata_go_proto"
 	"github.com/openconfig/ondatra"
 )
 
+// lookupDutDeviations returns the deviations for the specified dut. If no duts match `nil` is returned.
+func lookupDUTDeviations(dut *ondatra.DUTDevice) *mpb.Metadata_Deviations {
+	for _, platformExceptions := range metadata.Get().PlatformExceptions {
+		vendor := platformExceptions.GetPlatform().Vendor.String()
+		for _, hardwareModel := range platformExceptions.GetPlatform().HardwareModel {
```

Comment:
Check the vendor first, because then you can avoid the whole loop

Reply:
done